cli/compose/interpolation/interpolation.go

```diff
@@ -38,9 +39,13 @@ func interpolateSectionItem(
 	for key, value := range item {
 		interpolatedValue, err := recursiveInterpolate(value, mapping)
 		if err != nil {
+			hint := ""
+			if strings.Count(err.Template, "$") != 2*strings.Count(err.Template, "$$") {
```

Comment:
Thank you for the review. Just updated the PR. Looking at that check again, I think it also had the issue of being a bit too "low" level for the code that was around it.